backend/internal/system/cmodels/property.go

```diff
@@ -43,34 +90,64 @@ func (p *Property) SetEncrypted(encrypted bool) {
 	p.isEncrypted = encrypted
 }
 
+// GetStorageValue returns the value as is
+func (p *Property) GetStorageValue() string {
+	return p.value
+}
+
 // GetValue returns the decrypted value if it's a secret, otherwise returns the plain value
 func (p *Property) GetValue() (string, error) {
-	if !p.IsSecret || !p.IsEncrypted() {
-		return p.Value, nil
+	if !p.IsSecret() || !p.IsEncrypted() {
+		return p.value, nil
 	}
 
 	cryptoService := crypto.GetCryptoService()
-	decryptedValue, err := cryptoService.DecryptString(p.Value)
+	decryptedValue, err := cryptoService.DecryptString(p.value)
 	if err != nil {
-		return "", fmt.Errorf("failed to decrypt secret property %s: %w", p.Name, err)
+		return "", fmt.Errorf("failed to decrypt secret property %s: %w", p.GetName(), err)
 	}
 
 	return decryptedValue, nil
 }
 
 // Encrypt encrypts the value if it's a secret property
 func (p *Property) Encrypt() error {
-	if !p.IsSecret || p.Value == "" || p.IsEncrypted() {
+	if !p.IsSecret() || p.value == "" || p.IsEncrypted() {
 		return nil
 	}
 
 	cryptoService := crypto.GetCryptoService()
-	encryptedValue, err := cryptoService.EncryptString(p.Value)
+	encryptedValue, err := cryptoService.EncryptString(p.value)
 	if err != nil {
-		return fmt.Errorf("failed to encrypt secret property %s: %w", p.Name, err)
+		return fmt.Errorf("failed to encrypt secret property %s: %w", p.GetName(), err)
 	}
 
-	p.Value = encryptedValue
+	p.value = encryptedValue
 	p.SetEncrypted(true)
 	return nil
 }
+
+// ToProperty converts PropertyDTO to Property.
+// Returns an error if encryption fails for secret properties.
+func (dto *PropertyDTO) ToProperty() (*Property, error) {
+	return NewProperty(dto.Name, dto.Value, dto.IsSecret)
+}
+
+// ToPropertyDTO converts Property to PropertyDTO .
+func (p *Property) ToPropertyDTO() *PropertyDTO {
+	value, err := p.GetValue()
+	if err != nil {
+		log.GetLogger().Error("failed to get value for property", log.String("property", p.GetName()), log.Error(err))
```

Comment:
Let's return the error here also. Ideally if there's an error while constructing any of the property structs, it should be returned appropriately (either from svc layer or API).

This specific implementation will lead to returning a empty value leading to confusions.